test/functional/mempool_packages.py

```diff
@@ -58,6 +59,7 @@ def chain_transaction(self, node, parent_txid, vout, value, fee, num_outputs):
 
     def run_test(self):
         # Mine some blocks and have them mature.
+        self.nodes[0].add_p2p_connection(P2PTxInvStore()) # keep track of invs
```

Comment:
You still need to provide the callback. The default one does not reply with a getdata. See

```py
class P2PTxInvStore(P2PInterface):
    """A P2PInterface which stores a count of how many times each txid has been announced."""
    def __init__(self):
        super().__init__()
        self.tx_invs_received = defaultdict(int)

    def on_inv(self, message):
        # Store how many times invs have been received for each tx.
        for i in message.inv:
            if i.type == MSG_TX:
                # save txid
                self.tx_invs_received[i.hash] += 1

Reply:
You can run `git grep -W 'def on_inv'` to find examples on how to do that.

Reply:
Hmm. Looks like the tests passed on travis. I am still unsure why.

Reply:
Gah, I made a super dumb assumption that it would automatically call `P2PInterface`'s `on_inv` (which responds with the `getdata`) since it inherits from it. I'm working on that right now. Not sure why travis passed haha

Reply:
I reset travis, so that it properly shows failure. :sweat_smile:

Reply:
I spoke with @amitiuttarwar and decided to just have `P2PTxInvStore` call `super().on_inv` since she's planning on doing it anyway (in [#18807](https://github.com/bitcoin/bitcoin/pull/18807/files#diff-a77567bbc57dc1bbc033560e5cdd3b7dR647)). We've agreed to resolve any conflicts that happen depending on which gets merged first.